snet_cli/utils_ipfs.py

```diff
@@ -0,0 +1,30 @@
+# utils related to ipfs
+import tarfile
+import base58
+import glob
+import io
+import os
+
+
+# make tar from protodir/*proto, and publish this tar in ipfs
+# return base58 encoded ipfs hash
+def publish_proto_in_ipfs(ipfs_client, protodir):
+    files = glob.glob(os.path.join(protodir, "*.proto"))
+    
+    # We are sorting files before we add them to the .tar since an archive containing the same files in a different
+    # order will produce a different content hash;
+    files.sort()
```

Comment:
May be I missed something, but it seems that ```files``` collection contains full paths which in turn has prefix passed by user in command line. So it would be more correct to sort only basenames as only they are packed into tarball.

Reply:
All files are from the same directory. I don't support protobuf for several directories for now.